src/awesome_map/BoundaryInterceptor.js

```diff
@@ -232,10 +322,18 @@ define(function(require) {
                 this._touchState = targetState;
                 break;
 
-            case EventTypes.DRAG:
             case EventTypes.DRAG_END:
+                /* jshint -W086 */// Expected break statement 
+
+                this._debouncedDetermineBoundaryVisibility();
+                // Fall through
+
+            case EventTypes.DRAG:
             case EventTypes.DRAG_START:
+                /* jshint +W086 */// Expected break statement             
 
+                this._updateBoundariesStillVisible();
```

Comment:
Initially i had it debounce on all drag events as well, like it currently does for all ScrollEvents, but drag event behavior was a little different than scroll behavior, so i had to approach it a little bit differently.  

When dragging, a start_drag event gets fired, then some number of drag events fire until the drag is released after which drag_end fires.  Because of this, it was possible to drag down past the top of the page (which shouldnt neccesarily fire a boundary event, if we hadnt reached the top before dragging), and then stop and hold the drag (not releasing, but not moving either) which stops all of the events from firing, which means the boundary update debouncer resolves even though the user hasnt *technically* stopped dragging yet.  This led to weird things, like BoundaryEvents being fired when the user continued or let go of the drag.  So now it only re-checks visible boundaries on drag_END specifically.